data/governance/validations.go

```diff
@@ -0,0 +1,333 @@
+	if err != nil || !ok {
+		return false, err
+	}
+	// Verify Min Funding
+	ok, err = verifyMinFunding(config.InitialFunding, opt.ConfigUpdate.FundingGoal)
+	if err != nil || !ok {
+		return false, err
+	}
+	ok, err = verifyMinFunding(code.InitialFunding, opt.CodeChange.FundingGoal)
+	if err != nil || !ok {
+		return false, err
+	}
+	ok, err = verifyMinFunding(general.InitialFunding, opt.General.FundingGoal)
+	if err != nil || !ok {
+		return false, err
+	}
+	// Verify Funding and Voting Deadlines
+	if !verifyRangeInt64(config.FundingDeadline, minDeadlineFundingConfig, maxDeadlineFundingConfig) {
+
+		return false, errors.New("funding deadline for Config update is not within range")
+	}
+	if !verifyRangeInt64(config.VotingDeadline, minDeadlineVotingConfig, maxDeadlineVotingConfig) {
+		return false, errors.New("funding deadline for Config update is not within range")
+	}
+	if !verifyRangeInt64(code.FundingDeadline, minDeadlineFundingCode, maxDeadlineFundingCode) {
+		return false, errors.New("funding deadline for code update is not within range")
+	}
+	if !verifyRangeInt64(code.VotingDeadline, minDeadlineVotingCode, maxDeadlineVotingCode) {
+		return false, errors.New("funding deadline for code update is not within range")
+	}
+	if !verifyRangeInt64(general.FundingDeadline, minDeadlineFundingGeneral, maxDeadlineFundingGeneral) {
+		return false, errors.New("funding deadline for general update is not within range")
+	}
+	if !verifyRangeInt64(general.VotingDeadline, minDeadlineVotingeGeneral, maxDeadlineVotingGeneral) {
+		return false, errors.New("funding deadline for general update is not within range")
+	}
+	if !verifyRangeInt64(int64(config.PassPercentage), minPassPercentage, maxPassPercentage) {
+		return false, errors.New("pass percentage for config update is not within range")
+	}
+	if !verifyRangeInt64(int64(code.PassPercentage), minPassPercentage, maxPassPercentage) {
+		return false, errors.New("pass percentage for code update is not within range")
+	}
+	if !verifyRangeInt64(int64(general.PassPercentage), minPassPercentage, maxPassPercentage) {
+		return false, errors.New("pass percentage for general update is not within range")
+	}
+	ok, err = verifyDistribution(config.PassedFundDistribution)
+	if err != nil || !ok {
+		return false, err
+	}
+	ok, err = verifyDistribution(config.FailedFundDistribution)
+	if err != nil || !ok {
+		return false, err
+	}
+	ok, err = verifyDistribution(code.PassedFundDistribution)
+	if err != nil || !ok {
+		return false, err
+	}
+	ok, err = verifyDistribution(code.FailedFundDistribution)
+	if err != nil || !ok {
+		return false, err
+	}
+	ok, err = verifyDistribution(general.PassedFundDistribution)
+	if err != nil || !ok {
+		return false, err
+	}
+	ok, err = verifyDistribution(general.FailedFundDistribution)
+	if err != nil || !ok {
+		return false, err
+	}
+	if code.ProposalExecutionCost != oldOptions.CodeChange.ProposalExecutionCost {
```

Comment:
"me" seems to be a typo

Reply:
and all those sentence doesn't make sense

Reply:
You mean the error statements ? I just added Oneledger Address ,to signify that it is something that is set by us.I can change it to something else ,do you mean like " Bounty Address cannot be changed " ,

Reply:
it's better now.